fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/data/SavingsAccountDataValidator.java

```diff
@@ -415,6 +418,29 @@ private void throwExceptionIfValidationWarningsExist(final List<ApiParameterErro
         }
     }
 
+    private void validateLienParams(final DataValidatorBuilder baseDataValidator, final JsonElement element) {
+        if (this.fromApiJsonHelper.parameterExists(lienAllowedParamName, element)) {
+            final Boolean lienAllowed = this.fromApiJsonHelper.extractBooleanNamed(lienAllowedParamName, element);
+            baseDataValidator.reset().parameter(lienAllowedParamName).value(lienAllowed).ignoreIfNull().validateForBooleanValue();
+        }
+
+        if (this.fromApiJsonHelper.parameterExists(maxAllowedLienLimitParamName, element)) {
```

Comment:
As per the design, if lienAllowed is true, then max limit is mandatory. So instead of parameter exist,  we can check if lienAllowed.